### PR TITLE
[Feature] support regex_extact_all

### LINKED ISSUE #19815: [function]regexp_extract_all
besides regexp_extract, we need a function to extract all matched substring. Here is a proposal:

https://docs.starrocks.io/en-us/latest/sql-reference/sql-functions/string-functions/regexp_extract

the behavior is the same as regexp_ectract, but extract all the matched substrings into an array.
if no substring is matched, return an empty array.

### PR DESCRIPTION
**Pull request by Seaven**
Fixes https://github.com/StarRocks/starrocks/issues/19815

And fix `longest match`
before:
```
MySQL td> select regexp_extract("aEbECEdE", "(.*?E)", 1);
+-----------------------------------------+
| regexp_extract('aEbECEdE', '(.*?E)', 1) |
+-----------------------------------------+
| aEbECEdE                                |
+-----------------------------------------+
```

after:
```
MySQL td> select regexp_extract("aEbECEdE", "(.*?E)", 1);
+-----------------------------------------+
| regexp_extract('aEbECEdE', '(.*?E)', 1) |
+-----------------------------------------+
| aE                                      |
+-----------------------------------------+
```

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/sql-reference/sql-functions/like_predicate-functions/regexp_extract_all.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# regexp_extract
+
+## Description
+
+This function return all matching substrings in the target value which matches the regular expression pattern. It extracts the item in pos that matches the pattern. The pattern must completely match some parts of str so that the function can return parts needed to be matched in the pattern. If no matches are found, it will return an empty string.
+
+## Syntax
+
+```Haskell
+ARRAY<VARCHAR> regexp_extract_all(VARCHAR str, VARCHAR pattern, int pos)
```
Comment (fzhedu) [inline docs/sql-reference/sql-functions/like_predicate-functions/regexp_extract_all.md:10]: what is result from null?

Comment (Seaven) [reply docs/sql-reference/sql-functions/like_predicate-functions/regexp_extract_all.md:10]: is `null`

**Review by stdpain — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Comment by Seaven** [comment]
@mergify backport branch-3.1

**Pull request merged**